pyvista/plotting/picking.py

```diff
@@ -276,16 +276,24 @@ def enable_point_picking(self, callback=None, show_message=True,
             is specified as fraction of rendering window
             size. Rendering window size is measured across diagonal.
 
+        pickable_window : bool, optional
+            When True, points in the 3D window are pickable. Default to True.
+
         **kwargs : dict, optional
             All remaining keyword arguments are used to control how
             the picked point is interactively displayed.
 
         """
 
         def _end_pick_event(picker, event):
+
+            picked_point_id = picker.GetPointId()
+            if not pickable_window and not picked_point_id >= 0:
+                return
```

Comment:
I think this needs some robust user testing to make sure it actually works in practice and how it handles non-pickable meshes. When I was fiddling with this, there was some flaky behavior

Reply:
Good point. I've added a test that tries to pick a point near a pickable sphere (and succeeds) and then near a non-pickable sphere (and fails)